application/event_processor/changes.py

```diff
@@ -99,3 +100,16 @@ def update_changes_with_opening_times(changes: dict, dos_service: DoSService, nh
     if dos_std_open_dates != nhs_std_open_dates:
         logger.debug(f"Standard weekly opening times not equal. dos={dos_std_open_dates} and nhs={nhs_std_open_dates}")
         changes[OPENING_DAYS_KEY] = nhs_std_open_dates.export_cr_format()
+
+
+def update_changes_with_postcode(changes: dict, dos_service: DoSService, nhs_entity: NHSEntity) -> None:
+
+    dos_postcode = dos_service.postcode
+    nhs_postcode = nhs_entity.Postcode
+
+    if dos_postcode != nhs_postcode:
```

Comment:
My previous conversations with Nicole would suggest that we do a 'dumb' comparison between the exact values when checking to see if DoS needs to be updated, because we want the records to be identical. So for example we also do a 'dumb' check for the website and phone fields rather than checking for equivalence. 

Whereas when validating the postcode, a 'smart' check made more sense as a mis-formatted postcode is the same postcode still. However this still may not be considered valid when passed to DoS.

As these are business logic concerns I will check both of these points with Nicole.

Reply:
cool thanks Sam, apologies wasn't aware of that, happy if you and Nicole are happy

Reply:
@ed-sparkes We hashed it out and she has decided that

- Comparison between nhs and dos postcodes should be a 'smart' check. So no change request if they are the same but formatted differently.

- Validation is also a smart check, but the nicely formatted version found in the DoS DB is what will be put into the change request.

I also mentioned the website and phone fields have similar issues. She said she was happy for those to remain 'dumb' checks.